### PR TITLE
fix: Align search box with button

### PR DESCRIPTION
**Pull request by jimleroyer**
# Summary | Résumé

[The search button is not aligned with the search input box](https://trello.com/c/VEzLMQz9/528-misalignment-of-search-button-and-input-field):

image

The quick fix would have been to re-adjust the button top space that was previously defined in pixel. But I noticed from the start that the components were properly aligned with me because of my default font-size settings. This means that readjusting the top space would just move the problem to another font-size configuration and potentially affected impaired users with custom settings for accessibility reason.

Here is a screenshot of the button and search bar with a slightly increased font-size. The button is aligned for the bottom part but the top part isn't. The button with default style is not of the proper dimension for proper alignment:

image

## The long-term fix

The proper and longer fix, included in these changes, is to reconstruct the HTML elements to have the search button and search input bar in the same grid row, instead of trying to separately fix the two independently with their height. 

With the same previously increased font-size, the search button and input text with the fix looks better:

image

This overrides the default button styling though. I didn't want to change the default styling as we are due in the short term with upcoming design requests. We can clean up the overrides when that happens. 

## Find services by name admin screen

image

## Find users by email admin screen

image

## Responsiveness

The **old** implementation with a mobile view:

image

The **new** implementation with a mobile view:

image

## Accessibility

Although the accessibility of the search input text and button are sub par to our current desire, they remain unchanged in these changes. We will be addressing these soon as part of our a11y effort. 

# Test instructions | Instructions pour tester la modification

1. Head to the heroku instance.
2. Login and select a service to check.
3. From the service dashboard, select the sent emails to get to the proper screen to validate.

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

> Are there any related issues or tangent features you consider out of scope for
> this issue that could be addressed in the future?

---

> Y a-t-il des questions connexes que vous considérez hors sujet et qui
> pourraient être abordées plus tard?

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

**Review by e-pang — APPROVED** [review]
Thanks updating this! It's a small change but makes a lot of difference in making the service more refined.

File: app/templates/views/notifications.html [hunk 1]
```diff
@@ -37,16 +37,12 @@ <h1 class="heading-large">
       action=url_for('.view_notifications', service_id=current_service.id, message_type=message_type),
       class="grid-row contain-floats"
     ) %}
-      <div class="md:w-3/4 float-left py-0 px-0 px-gutterHalf box-border {% if message_type == 'sms' %}extra-tracking{% endif %}">
-        {{ textbox(
+      <div class="w-full float-left py-0 px-0 px-gutterHalf box-border {% if message_type == 'sms' %}extra-tracking{% endif %}">
+        {{ textbox_search(
```
Comment (AntoineAugusti) [inline app/templates/views/notifications.html:41]: Could you use this new component in other places? It seems to be used elsewhere https://github.com/cds-snc/notification-admin/search?q=align-button-with-textbox

File: app/templates/components/textbox-search.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  hint=False,
+  highlight_tags=False,
+  autofocus=False,
+  colour_preview=False,
+  width='',
+  suffix=None,
+  safe_error_message=False,
+  rows=8
+) %}
+  <div
```
Comment (AntoineAugusti) [inline app/templates/components/textbox-search.html:13]: Is it possible to call the existing `textbox` component here? I feel like a lot of this stuff is copy/paste. Ideally we would reuse the existing component and wrap it in a container.

**Review by AntoineAugusti — CHANGES_REQUESTED** [review]

File: app/templates/components/textbox-search.html (hunk 2 above)
Comment (jimleroyer) [reply app/templates/components/textbox-search.html:13]: No we can't. The textbox html structure of the textbox had to be reworked to include the button. Notice though that `field` was reused.

File: app/templates/components/textbox-search.html [hunk 3]
```diff
@@ -0,0 +1,61 @@
+{% macro textbox_search(
```
[the remark this answers is not in the document]
Comment (jimleroyer) [reply app/templates/components/textbox-search.html]: Style has been removed and usage of the new component has been expanded. I attached screenshots to the text description in the two pages where the new macro is now used.

**Review by AntoineAugusti — APPROVED** [review]
🚢

**Comment by jimleroyer** [comment]
[Misalignment of search button and input field](https://trello.com/c/VEzLMQz9/528-misalignment-of-search-button-and-input-field)

**Pull request merged**